fetch/api/redirect/redirect-method.any.js

```diff
@@ -7,26 +7,62 @@
 // |opts.expectedBodyAsString|: the expected response body as a string. The
 // server is expected to echo the request body. The default is the empty string
 // if the request after redirection isn't POST; otherwise it's |opts.body|.
+// |opts.expectedRequestContentType|: the expected Content-Type of redirected
+// request.
 function redirectMethod(desc, redirectUrl, redirectLocation, redirectStatus, method, expectedMethod, opts) {
-  var url = redirectUrl;
-  var urlParameters = "?redirect_status=" + redirectStatus;
+  let url = redirectUrl;
+  let urlParameters = "?redirect_status=" + redirectStatus;
   urlParameters += "&location=" + encodeURIComponent(redirectLocation);
 
-  var requestInit = {"method": method, "redirect": "follow"};
+  let requestHeaders = {
+    "Content-Encoding": "Identity",
+    "Content-Language": "en-US",
+    "Content-Location": "foo",
+  };
+  let requestInit = {"method": method, "redirect": "follow", "headers" : requestHeaders};
   opts = opts || {};
-  if (opts.body)
+  if (opts.body) {
     requestInit.body = opts.body;
+  }
 
   promise_test(function(test) {
     return fetch(url + urlParameters, requestInit).then(function(resp) {
+      let expectedRequestContentType = "NO";
+      if (opts.expectedRequestContentType) {
+        expectedRequestContentType = opts.expectedRequestContentType;
+      }
+
       assert_equals(resp.status, 200, "Response's status is 200");
       assert_equals(resp.type, "basic", "Response's type basic");
-      assert_equals(resp.headers.get("x-request-method"), expectedMethod, "Request method after redirection is " + expectedMethod);
+      assert_equals(
+        resp.headers.get("x-request-method"),
+        expectedMethod,
+        "Request method after redirection is " + expectedMethod);
+      let hasRequestBodyHeader = true;
+      if (opts.expectedStripRequestBodyHeader) {
+        hasRequestBodyHeader = !opts.expectedStripRequestBodyHeader;
+      }
+      assert_equals(
+        resp.headers.get("x-request-content-type"),
+        expectedRequestContentType,
+        "Request Content-Type after redirection is " + expectedRequestContentType);
+      [
+        "content-encoding",
+        "content-language",
+        "content-location"
+      ].forEach(header => {
+        let xHeader = "x-request-" + header;
+        assert_equals(
+          resp.headers.get(xHeader) != "NO",
+          hasRequestBodyHeader,
```

Comment:
It would be better to assert the actual value. So either "NO" or the value from the _requestHeaders_ object.

Reply:
done